appconfig/yaml.go

```diff
@@ -15,25 +15,47 @@
 package appconfig
 
 import (
-	"errors"
+	"fmt"
 
 	"gopkg.in/yaml.v2"
 )
 
 // YAMLRemoteRefParser parses b as a YAML-encoded RemoteRef. It assumes all
 // parsing errors mean the content is not a RemoteRef.
 func YAMLRemoteRefParser(path string, b []byte) (*RemoteRef, error) {
-	var ref RemoteRef
-	if err := yaml.UnmarshalStrict(b, &ref); err != nil {
+	var maybeRef struct {
+		Remote *string `yaml:"remote"`
+		Path   *string `yaml:"path"`
+		Ref    *string `yaml:"ref"`
+	}
+
+	if err := yaml.UnmarshalStrict(b, &maybeRef); err != nil {
```

Comment:
curious: why not just `Unmarshal`?

Reply:
If there are any fields that aren't part of the `RemoteRef` structure, the parser assumes the file is _not_ a remote reference. Using `UnmarshalStrict` returns an error if there are unknown fields.